src/app/builds/build-details/build-details.component.ts

```diff
@@ -2,31 +2,37 @@ import { ConfigService } from './../../shared/config/config.service';
 import { AfterViewChecked, Component, OnDestroy, OnInit } from '@angular/core';
 import { ActivatedRoute } from '@angular/router';
 import { BuildService, MainLog } from 'api-client';
-import { WharfProject } from 'src/app/models/main-project.model';
 import { BuildStatus } from '../../models/build-status';
+import { Title } from '@angular/platform-browser';
+
+const enum Tabs {
+  LOGS = 0,
+  ARTIFACTS,
+}
 
 @Component({
   selector: 'wh-build-details',
   templateUrl: './build-details.component.html',
 })
 export class BuildDetailsComponent implements OnInit, OnDestroy, AfterViewChecked {
-  buildId: string;
-  project: WharfProject;
+  buildId: number;
   buildStatus?: BuildStatus;
   myData: any;
   source: EventSource;
   listener: any = null;
   logEvents: MainLog[] = [];
   container: HTMLElement;
+  activeTabIndex = 0;
 
   constructor(
     private route: ActivatedRoute,
     private buildService: BuildService,
-    private configService: ConfigService) { }
+    private configService: ConfigService,
+    private titleService: Title) { }
 
   ngOnInit(): void {
-    this.buildId = this.route.snapshot.paramMap.get('buildId');
-    this.buildService.buildBuildidGet(Number(this.buildId)).subscribe(build => {
+    this.buildId = Number(this.route.snapshot.paramMap.get('buildId'));
```

Comment:
Nit: I find adding a plus to whatever you want to cast to number is a nicer shorter syntax.

```suggestion
    this.buildId = +this.route.snapshot.paramMap.get('buildId');
```

Reply:
I find it harder to read, but that's maybe because I'm not as pro-JS as you two

Reply:
I prefer `Number` because:
- It is harder to miss when reading.
- Won't be confused for a malformed `+=`.
- Won't make anybody think that there was supposed to be anything before it, that was accidentally left out.
  - e.g. `this.buildId = 1 + blabla`

Although with both `Number` and `+` casting, there is the interesting side effect that going to e.g. `/project/1.2e1` will bring you to the project with ID 12.